cmd/swarm/upload.go

```diff
@@ -36,22 +36,67 @@ import (
 	"gopkg.in/urfave/cli.v1"
 )
 
+var (
+	MAX_STDIN_SIZE = 1024 * 1000 * 100
+)
+
+func swapStdin() (size int, path string) {
+	b := make([]byte, 1024)
+	size = 0
+
+	td, err := ioutil.TempDir("", "swarm-stdin")
+	if err != nil {
+		utils.Fatalf("Could not create temporary directory: %s", err)
+	}
+	path = filepath.Join(td, "data")
+	f, err := os.Create(path)
```

Comment:
Rather than creating a tempdir and then creating a file in the tempdir, just use [ioutil.TempFile](https://golang.org/pkg/io/ioutil/#TempFile) (which makes clean up easier).